coalib/core/CircularDependencyError.py

```diff
@@ -12,8 +13,24 @@ def __init__(self, node=None):
             The node that was encountered twice and closes the dependency
             circle.
         """
-        message = 'Circular dependency detected.'
+        prefix = 'Circular dependency detected'
         if node is not None:
-            message += ' {0!r} -> ... -> {0!r}'.format(node)
+            message = '{prefix}: {node!r} -> ... -> {node!r}'.format(
+                prefix=prefix, node=node)
+        else:
+            message = prefix + '.'
+        return cls(message)
 
-        Exception.__init__(self, message)
+    @classmethod
+    def for_bears(cls, bears):
+        """
+        Creates the CircularDependencyError with a helpful message about the
+        dependency.
+
+        :param bears:
+            sequence of bears, for which a circular dependency was detected.
+        """
+        bear_names = [bear.name for bear in bears]
+
+        return cls('Circular dependency detected: {names}'.format(
+            names=' -> '.join(bear_names)))
```

Comment:
@zimmerman probably meant this comment https://github.com/coala/coala/pull/4199#discussion_r115360889 

@Makman2, I don't see any benefit (except beeing a bit more compact) in merging the two `for_*` clsmethods into `__init__`.  In https://github.com/coala/coala/pull/4199#discussion_r115496791 you move the logic to the client of the `CircularDependencyError` class, whereas in my code, the logic is part of the `CircularDependencyError` class.

I don't get why we need/want to optimize this code, since the exception class is only instantiated ecause (and immediately raised) in two places (Graphs.py, Dependencies.py) in the codebase (excluding tests).